test/extensions/coin-machine.js

```diff
@@ -382,6 +392,56 @@ contract("Coin Machine", (accounts) => {
       expect(balance).to.eq.BN(maxPerPeriod);
     });
 
+    it("cannot buy more than ther user limit allows", async () => {
```

Comment:
As far as I can tell, this test still passes even if you remove the checks on the limits in the contracts? So it's not really testing what it claims :stuck_out_tongue: